extensions/amp-experiment/1.0/attribute-allow-list/allowed-attribute-mutation-entry.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../../src/url';
+
+/**
+ * Base class for allowed attribute assertions.
+ * This should be extended to ensure the service
+ * is properly handled
+ */
+export class AllowedAttributeMutationEntry {
+  /**
+   * Function to validate the value
+   * for the attribute change. Subclasses
+   * may override.
+   * @param {!Object} mutationRecord
+   * @return {boolean}
+   */
+  validate(mutationRecord) {
```

Comment:
Is the `AllowedAttributeMutationEntry` base class's `#validate ()` method used by any extended class? 
Given there're only two methods, and each extended class is likely going to implement its own version of the `#validate()` function. I start to feel that using interface class, and have each class implement it might be a better solution compare to extend on top of base. 

I don't want to block your following PR. I'll leave the decision to you : )

Reply:
Accidentally clicked this when working on side projects. So I'm replying so I remember Tuesday :joy: 

But TL;DR Yup, class mutations will use validate. But I'm okay with an interface :+1: Let's chat Tuesday as I can explain how some override on or the other :smile: